tests/unit/device.ts

```diff
@@ -377,6 +377,30 @@ describe('Device', function() {
           sinon.assert.calledOnce(setupStreamSpy);
         });
 
+        it('should not throw during re-registration', async () => {
+          await registerDevice();
+
+          const regCalls: Array<Promise<any>> = [];
+
+          device.register = () => {
+            const regCall = Device.prototype.register.call(device);
+            regCalls.push(regCall);
+            return regCall;
+          };
+
+          device.updateOptions({ edge: 'sydney' });
+
+          pstream.emit('offline');
+          await clock.tickAsync(0);
+          pstream.emit('connected', { region: 'EU_IRELAND' });
+          await clock.tickAsync(0);
+          pstream.emit('ready');
+          await clock.tickAsync(0);
```

Comment:
Can we make these clock ticks synchronous with real values instead of zeroes? That way, it's closer to the real behavior.

Reply:
Synchronous ticks won't fire off `async` functions properly, this test won't work with `clock.tick` due to the internals of `Device` using `async` functions. I use `0` because I just want the stack to run all immediate `async` functions.